views/js/runner/helpers/offlineJumpTable.js

```diff
@@ -0,0 +1,420 @@
+            clearJumpTable: function clearJumpTable() {
+                jumpTable = [];
+
+                return this;
+            },
+
+            /**
+             * Adds a new jump into the end of the jump table
+             *
+             * @param {String} partIdentifier
+             * @param {String} sectionIdentifier
+             * @param {String} itemIdentifier
+             */
+            addJump: function addJump(partIdentifier, sectionIdentifier, itemIdentifier) {
+                var self = this;
+
+                return new Promise(function(resolve) {
+                    var nextPosition = 'position' in self.getLastJump()
+                        ? self.getLastJump().position + 1
+                        : 0;
+
+                    jumpTable.push({
+                        item: itemIdentifier,
+                        part: partIdentifier,
+                        section: sectionIdentifier,
+                        position: nextPosition
+                    });
+
+                    resolve();
+                });
+            },
+
+            /**
+             * Jumps into a specific position inside the jump table. The jump entry in the given position
+             * will be the last element of the jump table, every other entry after this entry will get deleted
+             *
+             * @param {Integer} position
+             * @returns {offlineJumpTableFactory}
+             */
+            jumpTo: function jumpTo(position) {
+                jumpTable = jumpTable.filter(function(jump) {
+                    return jump.position <= position;
+                });
+
+                return this;
+            },
+
+            /**
+             * Adds the next item to the end of the jump table
+             */
+            jumpToNextItem: function jumpToNextItem(params) {
+                var self = this;
+
+                addResponsesToResponseStore(params);
+
+                return new Promise(function(resolve) {
+                    var simplifiedTestMap = self._getSimplifiedTestMap();
+                    var lastJumpItem = self.getLastJump().item || null;
+                    var items = self._getItems();
+                    var itemSliceIndex = items.indexOf(lastJumpItem);
+                    var itemIdentifierToAdd = items.slice(itemSliceIndex + 1).shift();
+                    var itemToAdd = simplifiedTestMap
+                        .filter(function(row) {
+                            return row.item === itemIdentifierToAdd;
+                        })
+                        .shift();
+
+                    var lastJumpItemData = simplifiedTestMap
+                        .filter(function(row) {
+                            return row.item === lastJumpItem;
+                        })
+                        .shift();
+
+                    if (lastJumpItemData && lastJumpItemData.itemHasBranchRule) {
+                        return itemStore.get(lastJumpItem).then(function(item) {
```

Comment:
Please also catch the error, leaving promise uncaught might lead to inconsistent state.